File: solutions_5738606668808192_1/Python/zpc/google.py
import random

def random_str():
	result = '1'
	for i in range(N-2):
		result += str(random.randint(0,1))
	result += '1'
	
	return result

def legitimate(jamcoin):
	result = jamcoin
	for i in range(2,11):
		n = int(jamcoin, i)
		factor = findfactor(n)
		if 0 == factor:
			return None
		else:
			result += ' ' + str(factor)

	return result + '\n'

def findfactor(n):
	#assert primes[-1]*primes[-1]>n
	for p in primes:
		if p*p>n:
			return 0
		if n%p==0:
			return p
	return 0


if __name__ == '__main__':
	f = open('primes.txt','r')
	input = open('input.txt','r')
	output = open('output.txt','w')

	primes = [int(i) for i in f.read().split()]
	primes = primes[:1000]
	[T,N,J] = [int(i) for i in input.read().split()]
	
	j=0
	output.write('Case #1:\n')
	while(j<J):
		jamcoin = random_str()
		result = legitimate(jamcoin)
		if None == result:
			continue
		output.write(result)
		j+=1
	

File: solutions_5636311922769920_1/C/zpc/google.c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

int compute(int K, int C, int S, int *A, int *len)
{
	int i;
	if(S<K)return -1;
	for(i=0;i<S;i++){
		A[i] = i+1;
	}
	*len = S;
	
	return 0;
}

int main()
{
	int T,K,C,S,i,len,n,A[101],result;

	FILE *input = fopen("input.txt", "r");
	FILE *output = fopen("output.txt", "w");
	fscanf(input,"%d", &T);

	n=1;
	while(n<=T){
		fscanf(input, "%d%d%d", &K, &C, &S);
		//assert(K==S);
		result = compute(K,C,S,A,&len);

		if(-1==result){
			fprintf(output, "Case #%d: IMPOSSIBLE\n", n);
		}else{
			fprintf(output, "Case #%d:", n);
			for(i=0;i<len;i++)fprintf(output, " %d", A[i]);
			fprintf(output, "\n");
		}
		n++;
	}

	return 0;
}
